# GraspTask Parameter


# Equipment
equipment:
  - name: "Smart Sensor Box"
    robot_pick_pose:
      reference_frame: "base_footprint"
      position: {x: 0.03, y: 0.15, z: 0.925}
      orientation: {x: 0, y: 0, z: 0.7071, w: 0.7071}
    # robot_set_pose: #None
    release_on_set: True
    mesh_url: "package://tbf_gripper_tools/resources/mesh/wlan_box.stl"
    scale: {x: 1.0, y: 1.0, z: 1.0}
    pick_waypoints:
      pre_joint_values:   [-171.5, -137.2, 120.3, -163.53, -80.8, 45.2]
      grasp_joint_values: [-175.3, -114.9, 114.1, -179.50, -72.7, 45.3]
      post_joint_values:  [-175.3, -113.9, 103.5, -169.84, -72.8, 45.3]
  - name: "Water Sample Station"
    robot_pick_pose:
      reference_frame: "water_sample_station_mount_link"
      position: {x: -0.05, y: 0.105, z: -0.03}
      orientation: {x: -0.7071, y: 0, z: 0.0, w: 0.7071}
    # robot_set_pose: #None
    release_on_set: False
    mesh_url: "package://tbf_gripper_tools/resources/mesh/water_station_scaled.stl"
    scale: {x: 1.0, y: 1.0, z: 1.0}
    pick_waypoints:
      pre_joint_values:   [-12.18, -75.06, -31.19, -132.94, 80.34, 37.48]
      grasp_joint_values: [-13.50, -99.22, -35.34, -130.96, 80.41, 37.45]
      post_joint_values:  [-13.53, -107.3, -16.28, -141.95, 80.50, 37.48]

# Arm
arm:
  backup_joint_values: [-180, -90, 0.0, -90, 0.00, 0.0]
  home_joint_values:   [0.00, -90, 0.0, -90, 0.00, 0.0]
  watch_joint_values:  [-180, -90, -90, -80, 90.0, 125]
  linear_tcp_speed: 60.0
  linear_tcp_acceleration: 20.0
  joint_speed: 10.0
  joint_acceleration: 5.0
  joint_states_topic: "/ur5/joint_states"

# MoveIt!
moveit:
  eef_link: "gripper_robotiq_palm_planning"
  planner_id: "KPIECEkConfigDefault" # for parameter see the ompl_planning.yaml at julius_moveit
    # SBLkConfigDefault
    # ESTkConfigDefault
    # LBKPIECEkConfigDefault
    # BKPIECEkConfigDefault
    # KPIECEkConfigDefault
    # RRTkConfigDefault
    # RRTConnectkConfigDefault
    # RRTstarkConfigDefault
    # TRRTkConfigDefault
    # PRMkConfigDefault
    # PRMstarkConfigDefault
  planner_time: 20.0
  planner_attempts: 100
  reference_frame: "/gripper_ur5_base_link"
  workspace: [-1.6, -1.2, 0.0, 0.25, 1.2, 1.7]
  tolerance: 0.01
  planned_path_publisher: "julius_moveit/move_group/display_planned_path"
  group_name: "UR5"
  goal_tolerance: 0.05
  goal_position_tolerance: 0.01
  goal_orientation_tolerance: 0.01
  goal_joint_tolerance: 0.1
  allow_looking: False
  allow_replanning: False
  touch_links:
  - gripper_robotiq_palm
  - gripper_robotiq_finger_1_link_0
  - gripper_robotiq_finger_1_link_1
  - gripper_robotiq_finger_1_link_2
  - gripper_robotiq_finger_1_link_3
  - gripper_robotiq_finger_2_link_0
  - gripper_robotiq_finger_2_link_1
  - gripper_robotiq_finger-2_link_2
  - gripper_robotiq_finger_2_link_3
  - gripper_robotiq_finger_middle_link_0
  - gripper_robotiq_finger_middle_link_1
  - gripper_robotiq_finger_middle_link_2
  - gripper_robotiq_finger_middle_link_3